File: frontend/src/types/signal.ts
export type SignalType =
  | "funding_round"
  | "news"
  | "hiring_surge"
  | "github_activity"
  | "regulatory_change"
  | "environmental_alert"
  | "tender_deadline";

export interface Signal {
  id: string;
  companyId: string;
  type: SignalType;
  title: string;
  detail: string | null;
  amount: string | null;
  sourceUrl: string | null;
  detectedAt: string;
}

File: backend/models/signal.py
"""Signal model — news and funding signals per company."""

import uuid
from datetime import datetime
from sqlalchemy import String, Text, DateTime, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from db.session import Base


class Signal(Base):
    __tablename__ = "signals"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: f"sig_{uuid.uuid4().hex[:8]}")
    company_id: Mapped[str | None] = mapped_column(String, ForeignKey("companies.id"), nullable=True)
    type: Mapped[str] = mapped_column(String, nullable=False)       # 'funding_round' | 'news' | 'hiring_surge'
    title: Mapped[str | None] = mapped_column(Text, nullable=True)
    detail: Mapped[str | None] = mapped_column(Text, nullable=True)
    amount: Mapped[str | None] = mapped_column(String, nullable=True)
    source_url: Mapped[str | None] = mapped_column(Text, nullable=True)
    detected_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
